src/pudl/analysis/service_territory.py

```diff
@@ -536,77 +564,161 @@ def plot_all_territories(
 
 
 ################################################################################
-# Functions that provide a CLI to the service territory module
+# Provide a CLI for generating service territories
 ################################################################################
-def parse_command_line(argv):
-    """Parse script command line arguments. See the -h option.
+@click.command(
+    context_settings={"help_option_names": ["-h", "--help"]},
+)
+@click.option(
+    "--entity-type",
+    type=click.Choice(["util", "ba"]),
+    default="util",
+    show_default=True,
+    help=(
+        "What type of entity's service territories should be generated: Utility "
+        "(util) or Balancing Authority (ba)?"
+    ),
+)
+@click.option(
+    "--limit-by-state/--no-limit-by-state",
+    default=False,
+    help=(
+        "Limit service territories to including only counties located in states where "
+        "the utility or balancing authority also reported electricity sales in EIA-861 "
+        "in the year that the geometry pertains to. In theory a utility could serve a "
+        "county, but not sell any electricity there, but that seems like an unusual "
+        "situation."
+    ),
+    show_default=True,
+)
+@click.option(
+    "--year",
+    "-y",
+    "years",
+    type=click.IntRange(min=2001),
+    default=[],
+    multiple=True,
+    help=(
+        "Limit service territories generated to those from the given year. This can "
+        "dramatically reduce the memory and CPU intensity of the geospatial "
+        "operations. Especially useful for testing. Option can be used multiple times "
+        "toselect multiple years."
+    ),
+)
```

Comment:
Added the option to restrict the geometry compilation to a subset of the available years to keep resource usage down when necessary.